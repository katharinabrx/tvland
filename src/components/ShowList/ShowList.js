import React from "react";
import "./ShowList.scss";
import ShowTeaserCard from "../ShowTeaserCard";

import PlaceholderImage from "../../assets/placeholder.jpg";

const ShowList = ({ isLoading, showList }) => (
  <section className="ShowList">
    <h2>Shows</h2>
    <div className="ShowList__grid">
      {!isLoading ? (
        <React.Fragment>
          {showList.map((show, i) =>
            show.image === undefined ? (
              <ShowTeaserCard
                showId={show.id}
                showName={show.name}
                key={i}
                showImage={PlaceholderImage}
              />
            ) : (
              <ShowTeaserCard
                showId={show.id}
                showName={show.name}
                key={i}
                showImage={show.image.medium}
              />
            )
          )}
          )} }
        </React.Fragment>
      ) : (
        <React.Fragment>
          {[...Array(6)].map((_, i) => (
            <ShowTeaserCard key={i} showId={i} showName="Show" showImage={PlaceholderImage} />
          ))}
        </React.Fragment>
      )}
    </div>
  </section>
);
export default ShowList;
