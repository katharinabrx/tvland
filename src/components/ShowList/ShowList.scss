@import "../../styles/styles.scss";
.ShowList {
  padding: 1rem;
  @include breakpoint("sm") {
    padding: 2rem;
    grid-column-gap: 2rem;
    grid-template-columns: 1fr 2fr;
  }
  @include breakpoint("lg") {
    padding: 4rem;
    grid-column-gap: 4rem;
  }

  &__grid {
    display: grid;

    grid-template-columns: repeat(auto-fill, minmax(13.125rem, 1fr));
    //grid-template-columns: 1fr 1fr;

    grid-gap: 2rem 1rem;

    @include breakpoint("lg") {
      grid-gap: 3rem 2rem;
    }
  }

  &__link {
    text-decoration: none;
    width: 13.125rem;
    margin: 1rem;
  }

  &__card {
    width: 100%;
    height: 18.4375rem;
    margin-bottom: 1rem;

    @extend .card;
    text-decoration: none;
    &--placeholder {
      background-color: theme-color("background", "dark");
      height: 18.4375rem;
    }
  }
  &__title {
    color: theme-color("primary", "base");
    strong {
      text-decoration: none;
    }
  }
}
