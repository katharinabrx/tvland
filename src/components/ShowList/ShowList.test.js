import { shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";
import ShowList from "./ShowList";

it("expect to render ShowList component with placeholders", () => {
  const mockShowList = [];
  expect(toJson(shallow(<ShowList isLoading={true} showList={mockShowList} />))).toMatchSnapshot();
});

it("expect to render ShowList component", () => {
  const mockShowList = [
    {
      id: 1,
      name: "Show Name 1"
    },
    {
      id: 2,
      name: "Show Name 2"
    }
  ];
  expect(toJson(shallow(<ShowList isLoading={false} showList={mockShowList} />))).toMatchSnapshot();
});
