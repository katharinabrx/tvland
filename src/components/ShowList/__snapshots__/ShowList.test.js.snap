// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ShowList component 1`] = `
<section
  className="ShowList"
>
  <h2>
    Shows
  </h2>
  <div
    className="ShowList__grid"
  >
    <ShowTeaserCard
      key="0"
      showImage="placeholder.jpg"
      showName="Show"
    />
    <ShowTeaserCard
      key="1"
      showImage="placeholder.jpg"
      showName="Show"
    />
    <ShowTeaserCard
      key="2"
      showImage="placeholder.jpg"
      showName="Show"
    />
    <ShowTeaserCard
      key="3"
      showImage="placeholder.jpg"
      showName="Show"
    />
    <ShowTeaserCard
      key="4"
      showImage="placeholder.jpg"
      showName="Show"
    />
    <ShowTeaserCard
      key="5"
      showImage="placeholder.jpg"
      showName="Show"
    />
  </div>
</section>
`;

exports[`expect to render ShowList component with placeholders 1`] = `
<section
  className="ShowList"
>
  <h2>
    Shows
  </h2>
  <div
    className="ShowList__grid"
  >
    )} }
  </div>
</section>
`;
